File: sort/algorithms/algorithmsSort/BinaryTree/leetcode/searchBinaryTree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left    
        self.right = right

leftRightNode = TreeNode("4", None, None)
rightRightNode = TreeNode("5", None, None)
leftNode = TreeNode("2", None, leftRightNode)
rightNode = TreeNode("3", None, rightRightNode)
root = TreeNode("1", leftNode, rightNode)

 
def searchBST(myTree, val):
    if myTree is None:
        return None 
        
    if myTree.val == val:
        print("Found", myTree.val, val)
        return val
    
    if val < myTree.val:
        print("Left", myTree.val, val)
        return searchBST(myTree.left, val)
    else: 
        print("Right", myTree.val, val)
        return searchBST(myTree.right, val)

answer = searchBST(root, "2")
print(answer)



 

File: java/src/main/java/algorithms/BinaryTree/SearchBinaryTree.java
package algorithms.BinaryTree;

import java.util.ArrayList;

public class SearchBinaryTree {
    public static void main(String[] args) {
        ArrayList<String> list = new ArrayList<String>();

        TreeNode myTree = new TreeNode("4");
        TreeNode leftNode = new TreeNode("2");
        TreeNode rightNode = new TreeNode("7");
        TreeNode leftLeftNode = new TreeNode("1");
        TreeNode leftRightNode = new TreeNode("3");

        //Level 1
        myTree.left = leftNode;
        myTree.right = rightNode;

        //Level 2
        myTree.left.left = leftLeftNode;
        myTree.left.right = leftRightNode;

        TreeNode answer = searchBST(myTree, list);
        System.out.println(answer.val + " " + answer.left.val + " " + answer.right.val);

    }

    private static TreeNode searchBST(TreeNode root, ArrayList<String> list) {
        Integer val = 2;
        int rootVal = Integer.parseInt(root.val);

        if (root == null) {
            return null;
        }
        if (rootVal == val) {
            return root;
        }
        if (val < rootVal) {
            return searchBST(root.left, list);
        } else {
            return searchBST(root.right, list);
        }


    }

}
